sdk/core/performance-stress/src/program.cpp

```diff
@@ -14,26 +14,19 @@
 namespace {
 
 inline std::unique_ptr<Azure::PerformanceStress::PerformanceTest> PrintAvailableTests(
-    std::map<
-        std::string,
-        std::function<std::unique_ptr<Azure::PerformanceStress::PerformanceTest>(
-            Azure::PerformanceStress::TestOptions)>> const& tests)
+    std::vector<Azure::PerformanceStress::TestMetadata> const& tests)
 {
-  std::cout << "Available tests to run:" << std::endl;
+  std::cout << "No test name found in the input. Available tests to run:" << std::endl;
+  std::cout << std::endl << "Name\t\tDescription" << std::endl << "---\t\t---" << std::endl;
   for (auto test : tests)
   {
-    std::cout << "  - " << test.first << std::endl;
+    std::cout << test.Name << "\t\t" << test.Description << std::endl;
   }
   return nullptr;
 }
 
-inline std::function<std::unique_ptr<Azure::PerformanceStress::PerformanceTest>(
-    Azure::PerformanceStress::TestOptions)>
-GetTest(
-    std::map<
-        std::string,
-        std::function<std::unique_ptr<Azure::PerformanceStress::PerformanceTest>(
-            Azure::PerformanceStress::TestOptions)>> const& tests,
+inline Azure::PerformanceStress::TestMetadata const* GetTestMetadata(
+    std::vector<Azure::PerformanceStress::TestMetadata> const& tests,
```

Comment:
Store the tests as a Map as you had before.  Use the test name as the key, its ok to have the key in the map and in the TestMetadata.
The map is much more efficient for finding the tests we want to run.

Reply:
we don't really need that efficiency for finding the test to run. 
I am hoping to keep the perf program not knowing test details, as a test writer, I only want to take care of creating the test without worry too much about how do I add it to the perf program